BungeeCord-Patches/0049-Add-some-things-to-the-API.patch

```diff
@@ -0,0 +1,264 @@
+      *
+      * @param position the screen position
+      * @param message the message to send
++     * @deprecated This is unlikely the API you want to use. See {@link #sendActionBar(String)} or {@link #sendActionBar(BaseComponent)}, {@link #sendActionBar(BaseComponent...)} for a more proper ActionBar API.
+      */
++    @Deprecated // Waterfall
+     public void sendMessage(ChatMessageType position, BaseComponent message);
+ 
+     /**
+@@ -372,4 +376,37 @@ public interface ProxiedPlayer extends Connection, CommandSender
+      * @return this player's {@link Scoreboard}
+      */
+     Scoreboard getScoreboard();
++
++    // Waterfall start
++
++    /**
++     * Sends ActionBar message to the client.
++     *
++     * @param message the message you wish to send
++     */
++    public void sendActionBar(String message);
++
++    /**
++     * Sends ActionBar message to the client.
++     *
++     * @param message the message you wish to send
++     */
++    public void sendActionBar(BaseComponent message);
++
++    /**
++     * Sends ActionBar message to the client.
++     *
++     * @param message the message you wish to send
++     */
++    public void sendActionBar(BaseComponent... message);
++
++    /**
++     * Gets the current player title. The method can return null if the current
++     * title is null.
++     *
++     * @return title if not null
++     */
++    public Title getCurrentTitle();
++
++    // Waterfall end
+ }
+diff --git a/api/src/main/java/net/md_5/bungee/api/plugin/Plugin.java b/api/src/main/java/net/md_5/bungee/api/plugin/Plugin.java
+index 2e5ae4fb..981dd8ef 100644
+--- a/api/src/main/java/net/md_5/bungee/api/plugin/Plugin.java
++++ b/api/src/main/java/net/md_5/bungee/api/plugin/Plugin.java
+@@ -67,6 +67,13 @@ public class Plugin
+         return new File( getProxy().getPluginsFolder(), getDescription().getName() );
+     }
+ 
++    // Waterfall start - allow plugins to get plugin names directly
++    public String getName()
++    {
++        return description.getName();
++    }
++    // Waterfall end
++
+     /**
+      * Get a resource from within this plugins jar or container. Care must be
+      * taken to close the returned stream.
+diff --git a/api/src/main/java/net/md_5/bungee/api/plugin/PluginManager.java b/api/src/main/java/net/md_5/bungee/api/plugin/PluginManager.java
+index 81bd18f0..44ff7cd9 100644
+--- a/api/src/main/java/net/md_5/bungee/api/plugin/PluginManager.java
++++ b/api/src/main/java/net/md_5/bungee/api/plugin/PluginManager.java
+@@ -181,10 +181,14 @@ public class PluginManager
+             {
+                 if ( proxy.getConfig().isLogCommands() )
+                 {
+-                    proxy.getLogger().log( Level.INFO, "{0} executed command: /{1}", new Object[]
++                    // Waterfall - check if the command sender is not the console
++                    if ( !sender.getName().equalsIgnoreCase( proxy.getConsole().getName() ) )
```

Comment:
Logging console commands might be useful for web terminals/screen/tmux... where multiple people might be watching the same console output

Reply:
And generally useful when checking the log file?